azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -92,6 +92,8 @@ def decrypt_p7m(self, p7m_file, trans_prv_file, trans_cert_file, pem_file):
                 "-out", pem_file]
 
             first_proc = subprocess.Popen(first_cmd, stdout=subprocess.PIPE)
+            first_proc.wait()
```

Comment:
The wait has the potential to introduce a deadlock (different issue to the above comment): if the first command produces too much output the pipe may block until it is consumed, which will not happen until the next command is started, which will not happen until the wait completes.

Reply:
See https://docs.python.org/2/library/subprocess.html#replacing-shell-pipeline

The original code was missing the call to close()